### ISSUE TITLE
Pilot fails with "Your Apple ID or password was entered incorrectly" with valid login

### ISSUE THREAD

**Issue by the-mace** [fastlane/fastlane#9045]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description

Starting a couple weeks ago, Fastlane started failing with pilot with a "Your Apple ID or password was entered incorrectly". This is after it claims to successfully log into iTunes connect. Entering a valid email and password for iTunes connect does not work. Apple's Application Loader with the same account works fine and is how I've been working around it for now.

##### Complete output when running fastlane, including the stack trace and command used
> You can use: `--capture_output` as the last commandline argument to get that collected for you

Since the IPA creation is fine, pasting the output from IPA creation on:

[17:51:12]: Successfully exported and signed the ipa file:
[17:51:12]: /users/user/Documents/Code/hfc/tesla-mobile/Teslab.ipa
[17:51:12]: -------------------
[17:51:12]: --- Step: pilot ---
[17:51:12]: -------------------
[17:51:12]: Login to iTunes Connect (user@company.com)
[17:51:15]: Login successful
[17:51:15]: Ready to upload new build to TestFlight (App: 11459XXXXX)...
[17:51:16]: Going to upload updated app to iTunes Connect
[17:51:16]: This might take a few minutes. Please don't interrupt the script.
[17:51:21]: [Transporter Error Output]: Your Apple ID or password was entered incorrectly. (-20101)
-------------------------------------------------------------------------------------
Please provide your Apple Developer Program account credentials
The login information you enter will be stored in your macOS Keychain
You can also pass the password using the `FASTLANE_PASSWORD` environment variable
More information about it on GitHub: https://github.com/fastlane/fastlane/tree/master/credentials_manager
-------------------------------------------------------------------------------------
Username: user@company.com
The login credentials for 'user@company.com' seem to be wrong
Do you want to re-enter your password? (y/n)
y
Removing Keychain entry for user 'user@company.com'...
keychain: "/users/user/Library/Keychains/login.keychain-db"
version: 512
class: "inet"
attributes:
    0x00000007 <blob>="deliver.user@company.com"
    0x00000008 <blob>=<NULL>
    "acct"<blob>="user@company.com"
    "atyp"<blob>="dflt"
    "cdat"<timedate>=0x32303137303530323231313934325A00  "20170502211942Z\000"
    "crtr"<uint32>=<NULL>
    "cusi"<sint32>=<NULL>
    "desc"<blob>=<NULL>
    "icmt"<blob>=<NULL>
    "invi"<sint32>=<NULL>
    "mdat"<timedate>=0x32303137303530323231313934325A00  "20170502211942Z\000"
    "nega"<sint32>=<NULL>
    "path"<blob>=<NULL>
    "port"<uint32>=0x00000000 
    "prot"<blob>=<NULL>
    "ptcl"<uint32>=0x00000000 
    "scrp"<sint32>=<NULL>
    "sdmn"<blob>=<NULL>
    "srvr"<blob>="deliver.user@company.com"
    "type"<uint32>=<NULL>
password has been deleted.
-------------------------------------------------------------------------------------
Please provide your Apple Developer Program account credentials
The login information you enter will be stored in your macOS Keychain
You can also pass the password using the `FASTLANE_PASSWORD` environment variable
More information about it on GitHub: https://github.com/fastlane/fastlane/tree/master/credentials_manager
-------------------------------------------------------------------------------------
Password (for user@company.com): 
Password (for user@company.com): ***************
[17:53:10]: Please run this tool again to apply the new password
[17:53:10]: Transporter transfer failed.
[17:53:10]: 
[17:53:10]: Your Apple ID or password was entered incorrectly. (-20101)
[17:53:10]: Your Apple ID or password was entered incorrectly. (-20101)
Return status of iTunes Transporter was 1: Your Apple ID or password was entered incorrectly. (-20101)
The call to the iTMSTransporter completed with a non-zero exit status: 1. This indicates a failure.
+--------------------+--------------------------------------+
|                       Lane Context                        |
+--------------------+--------------------------------------+
| DEFAULT_PLATFORM   | ios                                  |
| PLATFORM_NAME      | ios                                  |
| LANE_NAME          | ios beta                             |
| VERSION_NUMBER     | 1.5.1                                |
| BUILD_NUMBER       | 300                                  |
| SIGH_PROFILE_TYPE  | app-store                            |
| IPA_OUTPUT_PATH    | /users/user/Documents/Code/hfc/tesla  |
|                    | -mobile/Teslab.ipa                   |
| DSYM_OUTPUT_PATH   | /users/user/Documents/Code/hfc/tesla  |
|                    | -mobile/Teslab.app.dSYM.zip          |
| XCODEBUILD_ARCHIVE | /users/user/Library/Developer/Xcode/  |
|                    | Archives/2017-05-02/Teslab           |
|                    | 2017-05-02 17.49.33.xcarchive        |
+--------------------+--------------------------------------+
[17:53:10]: Error uploading ipa file, for more information see above
+------+------------------------+-------------+
|              fastlane summary               |
+------+------------------------+-------------+
| Step | Action                 | Time (in s) |
+------+------------------------+-------------+
| 1    | Verifying required     | 0           |
|      | fastlane version       |             |
| 2    | default_platform       | 0           |
| 3    | Switch to ios          | 0           |
|      | set_version_specific   |             |
|      | lane                   |             |
| 4    | increment_version_num  | 2           |
|      | ber                    |             |
| 5    | Switch to ios          | 0           |
|      | build_version_bump     |             |
|      | lane                   |             |
| 6    | increment_build_numbe  | 2           |
|      | r                      |             |
| 7    | match                  | 6           |
| 8    | gym                    | 101         |
| 💥   | pilot                  | 118         |
+------+------------------------+-------------+
[17:53:10]: fastlane finished with errors
[!] Error uploading ipa file, for more information see above

### Environment

Please run `fastlane env` and copy the output below. This will help us help you :+1:
If you used `--capture_output` option please remove this block - as it is already included there.

<details><summary>✅ fastlane environment ✅</summary>
### Stack
| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.12.4                                     |
| Ruby                        | 2.2.4                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.6.4                           |
| Installation Source         | ~/.fastlane/bin/bundle/bin/fastlane         |
| Host                        | Mac OS X 10.12.4 (16E195)                   |
| Ruby Lib Dir                | ~/.fastlane/bin/bundle/lib                  |
| OpenSSL Version             | OpenSSL 1.0.2g  1 Mar 2016                  |
| Is contained                | false                                       |
| Is homebrew                 | true                                        |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 8.3.2                                       |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE | en_US.UTF-8 | ✅ |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# Customise this file, documentation can be found here:
# https://github.com/fastlane/fastlane/tree/master/fastlane/docs
# All available actions: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Actions.md
# can also be listed using the `fastlane actions` command

# Change the syntax highlighting to Ruby
# All lines starting with a # are ignored when running `fastlane`

# If you want to automatically update fastlane if a new version is available:
# update_fastlane

# This is the minimum version number required.
# Update this, if you use features of a newer version
fastlane_version '1.105.3'

default_platform :ios

platform :ios do
  desc 'Submit a new Beta Build to Testflight'
  lane :beta do
    match(type: 'appstore', app_identifier: 'com.company.tesla')
    ipa = gym(
      clean: true,
      scheme: 'Teslab',
      project: './ios/Teslab.xcodeproj'
    )

    pilot(skip_waiting_for_build_processing: true)
  end
end
```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
app_identifier 'com.happyfuncorp.tesla'
team_id 'XQZCRE565B'
itc_team_id "2317240"
apple_id ENV["USERNAME"]
json_key_file "./fastlane/googlePlay.json"
package_name 'com.happyfuncorp.tesla'
```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.28.7  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| slack-notifier            | 1.5.1        |
| CFPropertyList            | 2.3.5        |
| claide                    | 1.0.1        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.4.4        |
| rouge                     | 1.11.1       |
| xcpretty                  | 0.2.6        |
| terminal-notifier         | 1.7.1        |
| unicode-display_width     | 1.1.3        |
| terminal-table            | 1.7.3        |
| plist                     | 3.2.0        |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.1        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.5.0        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| highline                  | 1.7.8        |
| commander-fastlane        | 4.4.4        |
| excon                     | 0.55.0       |
| faraday                   | 0.12.1       |
| unf_ext                   | 0.0.7.4      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170404 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 2.1.0        |
| gh_inspector              | 1.0.3        |
| uber                      | 0.0.15       |
| representable             | 2.3.0        |
| retriable                 | 2.1.0        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| hurley                    | 0.2          |
| little-plugger            | 1.1.4        |
| multi_json                | 1.12.1       |
| logging                   | 2.2.2        |
| jwt                       | 1.5.6        |
| memoist                   | 0.15.0       |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.1        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.9.28       |
| json                      | 1.8.1        |
| mini_magick               | 4.5.1        |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 0.0.4        |
| dotenv                    | 2.2.0        |
| bundler                   | 1.14.6       |
| faraday_middleware        | 0.11.0.1     |
</details>

*generated on:* **2017-05-02**
</details>

**Comment by KrauseFx** [comment]
The error message is raised by the iTunes Transporter CLI:
```
[17:53:10]: Your Apple ID or password was entered incorrectly. (-20101)
[17:53:10]: Your Apple ID or password was entered incorrectly. (-20101)
Return status of iTunes Transporter was 1: Your Apple ID or password was entered incorrectly. (-20101)
The call to the iTMSTransporter completed with a non-zero exit status: 1. This indicates a failure.
```

Do you have 2 factor enabled? Do you have a password set in `DELIVER_PASSWORD` or `FASTLANE_PASSWORD` env variables? Does your password contain any special characters like a dollar symbol? Have you tried changing it to a password containing just characters and numbers, but having a longer one? 

We would love to fix this, however we'd need a way to reproduce it first 👍

**Comment by the-mace** [comment]
Hi, thanks for the fast response! 

1) 2 factor is not on on my iTunes Connect account.
2) I saw other bug reports about password characters so one of the things I tried (and the way my password still is) is without anything but alpha numeric.
3) No password set in FASTLANE_PASSWORD or DELIVER_PASSWORD. Should there be one?

**Comment by the-mace** [comment]
One thing I just thought of is that I'm working on multiple apps. Different directories but all using Fastlane. One of my other apps which I just deployed via Fastlane about the same time this app started breaking is on an account that has 2 factor authentication. 

I looked through my changes and found I had an environment variable, FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD, set for the other app. That had leaked into this environment and naturally doesn't work for this login. 

I removed it and now things are working again. Sorry for the troubles.

Is there a way to make this work from a pure Fastlane perspective and not be dependent on environment variables?
